packages/search-component/index.js

```diff
@@ -158,13 +165,20 @@ class Search extends Component {
               <dt class="f6 b ${this.local.artists.length ? 'db' : 'dn'}">Play history</dt>
               <dd class="ma0">
                 <ul class="list ma0 pa0 flex flex-column">
-                  ${this.local.artists.map(({ meta_value: name }) => html`
-                    <li class="mb2">
-                      <a class="db lh-copy f5 link" href="/search?q=${name}">
-                        ${name}
-                      </a>
-                    </li>
-                  `
+                  ${this.local.artists.map(({ meta_value: name }) => {
+                    const clickSearchHistoryHandler = () => {
+                      document.getElementById(`search-history-href-${name}`).click()
+                      this.state.components.header.machine.emit('search:toggle')
+                    }
```

Comment:
I don't understand what is going on here, we're just forwarding the click from the visible anchor to the invisible one? Can't we just use a single anchor to achieve all we want (putting an onclick on an anchor shouldn't disable its default behavior...I think?).

Reply:
Pressing href on a button redirects to the artist page or tag without reloading the header, whereas with an onClick with location=href, it does. If you have any ideas for achieving this (go to new page from an onclick without reloading the header too) I'm all ears! The UX impression wasn't as good with the header reloading (for no good reason) made the website seem slower overall.

Reply:
I assumed there was a way to navigate to a new page programmatically (as in, within the SPA framework) 😅 
I feel like I've seen an event you can emit perhaps (to which you pass the new path)? I can do some digging later, but I believe there is a way.

Reply:
Found it:
https://github.com/choojs/choo#pushstatestateeventspushstate

Reply:
I found this last night as well! Was too tired to implement. Will aim to get this working this evening. 🤩

Reply:
Here we go: https://github.com/resonatecoop/stream/pull/216/commits/424e5d2fc3c63f4e069a0f6394178724c1afd0a6

Reply:
Looks great now, sorry for the pickiness (not like I'm the authority anyway though) :smile:

Reply:
Not at all, this is a way less clunky implementation. I appreciate the review. 😀